### PR TITLE
Add test for the `$hidden` property

### LINKED ISSUE #1119: Does laravel-mongodb respect $hidden?
I'm having trouble figuring this out in the code, and I'm not sure if maybe I'm doing it wrong, but I have fields I'd like to add to a model that I don't want saved into the final document. Eloquent usually uses $hidden for this kind of stuff, but it doesn't seem to work here. Am I missing something?

### PR DESCRIPTION
**Pull request by Treggats**
As requested in https://github.com/mongodb/laravel-mongodb/issues/1119

closes #1119

This adds a test which covers the working of the `$hidden` property.

### CODE REVIEW

File: tests/PropertyTest.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Animal::truncate();
+        HiddenAnimal::truncate();
+    }
+
+    #[Test]
+    public function canHideCertainProperties(): void
```
Comment (alcaeus) [inline tests/PropertyTest.php]:
```suggestion
    public function testCanHideCertainProperties(): void
```

**Review by alcaeus — CHANGES_REQUESTED** [review]
Just a minor suggestion for consistency, but LGTM other than that.

File: tests/Models/Animal.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * @property string $name
+ * @property string $country
+ * @property bool $can_be_eaten
+ * @mixin Eloquent
+ * @method static Builder create(...$values)
+ * @method static Builder truncate()
+ * @method static Eloquent sole(...$parameters)
+ */
+final class Animal extends Eloquent
```
Comment (GromNaN) [inline tests/Models/Animal.php]: I don't see what is tested with this model class. Hidden fields and the `toArray` feature are tested with `HiddenAnimal`.

Comment (Treggats) [reply tests/Models/Animal.php]: The idea of the two classes is that one has the hidden property and another doesn't. As to prove the working of the hidden property.

Or do you suggest to have one class and use different fields to test?

File: tests/PropertyTest.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Eloquent;
+
+use MongoDB\Laravel\Tests\Models\Animal;
+use MongoDB\Laravel\Tests\Models\HiddenAnimal;
+use MongoDB\Laravel\Tests\TestCase;
+use PHPUnit\Framework\Attributes\Test;
```
Comment (Treggats) [inline tests/PropertyTest.php]: After the suggestion on line 25, this will become unused.
```suggestion
```

File: tests/Models/Animal.php (hunk 2 above)
Comment (GromNaN) [reply tests/Models/Animal.php]: IMHO, the test on HiddenAnimal already covers the feature.

Comment (Treggats) [reply tests/Models/Animal.php]: Fair enough, will remove this class and its usage

File: tests/PropertyTest.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Tests\Eloquent;
+
+use MongoDB\Laravel\Tests\Models\Animal;
```
Comment (Treggats) [inline tests/PropertyTest.php]:
```suggestion
```

File: tests/PropertyTest.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+use function assert;
+
+final class PropertyTest extends TestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        Animal::truncate();
```
Comment (Treggats) [inline tests/PropertyTest.php]:
```suggestion
```

File: tests/PropertyTest.php [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        HiddenAnimal::truncate();
+    }
+
+    public function testCanHideCertainProperties(): void
+    {
+        Animal::create([
+            'name' => 'Lion',
+            'country' => 'Central Africa',
+            'can_be_eaten' => false,
+        ]);
```
Comment (Treggats) [inline tests/PropertyTest.php]:
```suggestion
    {
```

File: tests/PropertyTest.php [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+        $animal = Animal::sole();
+        assert($animal instanceof Animal);
+        self::assertSame('Central Africa', $animal->country);
+        self::assertFalse($animal->can_be_eaten);
+
+        self::assertArrayHasKey('name', $animal->toArray());
+        self::assertArrayHasKey('country', $animal->toArray());
+        self::assertArrayHasKey('can_be_eaten', $animal->toArray());
```
Comment (Treggats) [inline tests/PropertyTest.php]:
```suggestion
        ]);
```

**Review by GromNaN — APPROVED** [review]

**Review by alcaeus — APPROVED** [review]

**Comment by GromNaN** [comment]
Thank you @Treggats

**Pull request merged**